WIP: Adding seed to the random number generation to allow reproducibility

kumaranu opened the pull request:
This PR is to put allow the user to provide random seed values to keep the results consistent while testing.

There are two instances of random number generation in the code:

1. geodesic_interpolate/interpolation.py
2. geodesic_interpolate/coord_utils.py

The goal here is to only provide the seed value when the user wants to.

virtualzx-nad reviewed (commented):
Thanks for adding reproducibility support! A few suggestions:

1. Consider using a single `--seed` parameter instead of two separate ones. The package can internally derive different seeds for different functions if needed.

2. The help text could be more concise, e.g.:
   ```python
   ps.add_argument("--seed", type=int, help="Random seed for reproducible results")
   ```

3. The implementation part is missing - need to pass these seeds to the respective functions and set numpy's random seed.

---
Review by [Augment](https://www.augmentcode.com/), approved by user.

kumaranu commented:
> Thanks for adding reproducibility support! A few suggestions:
> 
> 1. Consider using a single `--seed` parameter instead of two separate ones. The package can internally derive different seeds for different functions if needed.
> 2. The help text could be more concise, e.g.:
>    ```python
>    ps.add_argument("--seed", type=int, help="Random seed for reproducible results")
>    ```
> 3. The implementation part is missing - need to pass these seeds to the respective functions and set numpy's random seed.
> 
> Review by [Augment](https://www.augmentcode.com/), approved by user.

Hi @virtualzx-nad ,
I have added my response. Also added a unit test to confirm that the seed reproduces the result. I added a github actions workflow, which is waiting for some approval. Please check if you could approve that. Thanks.

kumaranu commented:
<img width="1940" height="186" alt="image" src="https://github.com/user-attachments/assets/8d42b18f-47ad-4a57-b11f-61d679fbae5d" />
Hi @virtualzx-nad,
I have added a workflow file to run a test for the seed's validity. But that file needs approval here. Thanks.
